heartbeat/docs/monitors/monitor-common-options.asciidoc

```diff
@@ -0,0 +1,202 @@
+
+Also see the <<monitors-scheduler,task scheduler>> settings.
+
+[float]
+[[monitor-ipv4]]
+==== `ipv4`
+
+A Boolean value that specifies whether to ping using the ipv4 protocol if
+hostnames are configured. The default is `true`.
+
+[float]
+[[monitor-ipv6]]
+==== `ipv6`
+
+A Boolean value that specifies whether to ping using the ipv6 protocol
+if hostnames are configured. The default is `true`.
+
+[float]
+[[monitor-mode]]
+==== `mode`
+
+If `mode` is `any`, the monitor pings only one IP address for a hostname. If
+`mode` is `all`, the monitor pings all resolvable IPs for a hostname. The
+`mode: all` setting is useful if you are using a DNS-load balancer and want to
+ping every IP address for the specified hostname. The default is `any`.
+
+[float]
+[[monitor-timeout]]
+==== `timeout`
+
+The total running time for each ping test. This is the total time allowed for
+testing the connection and exchanging data. The default is 16 seconds (16s).
+
+If the timeout is exceeded, {beatname_uc} publishes a `service-down` event. If the
+value specified for `timeout` is greater than `schedule`, intermediate checks
+will not be executed by the scheduler.
+
+[float]
+[[monitor-fields]]
+==== `fields`
+
+Optional fields that you can specify to add additional information to the
+output. For example, you might add fields that you can use for filtering log
+data. Fields can be scalar values, arrays, dictionaries, or any nested
+combination of these. By default, the fields that you specify here will be
+grouped under a `fields` sub-dictionary in the output document. To store the
+custom fields as top-level fields, set the `fields_under_root` option to true.
+If a duplicate field is declared in the general configuration, then its value
+will be overwritten by the value declared here.
+
+[float]
+[[monitor-fields-under-root]]
+==== `fields_under_root`
+
+If this option is set to true, the custom <<monitor-fields,fields>>
+are stored as top-level fields in the output document instead of being grouped
+under a `fields` sub-dictionary. If the custom field names conflict with other
+field names added by {beatname_uc}, then the custom fields overwrite the other
+fields.
+
+[float]
+[[monitor-tags]]
+==== `tags`
+
+A list of tags that will be sent with the monitor event. This setting is optional.
+
+[float]
+[[monitor-processors]]
+==== `processors`
+
+A list of processors to apply to the data generated by the monitor.
+
+See <<filtering-and-enhancing-data>> for information about specifying
+processors in your config.
+
+[float]
+[[monitor-keep-null]]
+==== `keep_null`
+
+If this option is set to true, fields with `null` values will be published in
+the output document. By default, `keep_null` is set to `false`.
+
+//REVIEWERS: Can we remove watch.poll_file since it was deprecated in 6.x?
+
+[float]
```

Comment:
I'm totally OK with your suggestion to remove this given how long ago it was deprecated, and that no one has asked any questions about it in forever. Having it here is probably just confusing.